pkg/storage/stores/shipper/storage/prefixed_object_client.go

```diff
@@ -0,0 +1,55 @@
+package storage
+
+import (
+	"context"
+	"io"
+	"strings"
+
+	"github.com/grafana/loki/pkg/storage/chunk"
+)
+
+type prefixedObjectClient struct {
+	downstreamClient chunk.ObjectClient
```

Comment:
nit: you could embed and only overrides function you want to overrides.

Reply:
I did it this way deliberately because if we add new methods then we could miss adding the prefix. Without handling the new methods here the build would fail so we would know we need to add them.